cpp/src/quantiles/tdigest/tdigest_aggregation.cu

```diff
@@ -851,30 +1016,35 @@ std::unique_ptr<column> group_merge_tdigest(column_view const& input,
   // generate the merged (but not yet compressed) tdigests for each group.
   std::vector<std::unique_ptr<table>> tdigests;
   tdigests.reserve(num_groups);
-  std::transform(
-    h_outer_offsets.begin(),
-    h_outer_offsets.end() - 1,
-    std::next(h_outer_offsets.begin()),
-    std::back_inserter(tdigests),
-    [&](auto tdigest_start, auto tdigest_end) {
-      // the range of tdigests in this group
-      auto const num_tdigests = tdigest_end - tdigest_start;
-
-      // slice each tdigest from the input
-      std::vector<table_view> unmerged_tdigests;
-      unmerged_tdigests.reserve(num_tdigests);
-      auto offset_iter = std::next(h_inner_offsets.begin(), tdigest_start);
-      std::transform(offset_iter,
+  std::transform(h_outer_offsets,
+                 h_outer_offsets + num_groups,
+                 std::next(h_outer_offsets),
+                 std::back_inserter(tdigests),
+                 [&](auto tdigest_start, auto tdigest_end) {
+                   // the range of tdigests in this group
+                   auto const num_tdigests = tdigest_end - tdigest_start;
+
+                   // slice each tdigest from the input
+                   std::vector<table_view> unmerged_tdigests;
+                   unmerged_tdigests.reserve(num_tdigests);
+                   auto offset_iter = std::next(h_inner_offsets.begin(), tdigest_start);
+                   std::transform(
+                     offset_iter,
                      offset_iter + num_tdigests,
                      std::next(offset_iter),
                      std::back_inserter(unmerged_tdigests),
-                     [&](auto start, auto end) {
+                     [&](size_type start, size_type end) {
                        return cudf::detail::slice(tdigests_unsliced, {start, end}, stream);
                      });
 
-      // merge
-      return cudf::detail::merge(unmerged_tdigests, {0}, {order::ASCENDING}, {}, stream, mr);
-    });
+                   // merge
+                   return cudf::detail::merge(unmerged_tdigests,
+                                              {0},
+                                              {order::ASCENDING},
+                                              {},
+                                              stream,
+                                              rmm::mr::get_current_device_resource());
```

Comment:
Why was this changed from the passed in memory resource?

Reply:
This is just a temporary, so it should be using the default allocator.  Same with the concatenated `merged` variable below.